R/serializer.R

```diff
@@ -497,26 +501,37 @@ self_set_serializer <- function(self, serializer) {
 #' The graphics device `dev_on` function will receive any arguments supplied to the serializer in addition to `filename`.
 #' `filename` points to the temporary file name that should be used when saving content.
 #' @param dev_off Function to turn off the graphics device. Defaults to [grDevices::dev.off()]
+#' @param serializer_params More docs needed here
+#'
 #' @export
-serializer_device <- function(type, dev_on, dev_off = grDevices::dev.off) {
+serializer_device <- function(type, dev_on, dev_off = grDevices::dev.off, serializer_params = list()) {
 
   stopifnot(!missing(type))
 
   stopifnot(!missing(dev_on))
   stopifnot(is.function(dev_on))
   stopifnot(length(formals(dev_on)) > 0)
   if (!any(c("filename", "...") %in% names(formals(dev_on)))) {
-    stop("`dev_on` must contain an arugment called `filename` or have `...`")
+    stop("`dev_on` must contain an argument called `filename` or have `...`")
   }
 
+  dev_requires_req <- "req" %in% names(formals(dev_on))
```

Comment:
Adding validations on `dev_on()`

** Untested

```suggestion
  dev_on_arg_names <- names(formals(dev_on))
  dev_on_requires_req <- "req" %in% dev_on_arg_names
  # Make sure extra args exist
  if (dev_on_requires_req) {
    dots_pos <- which("..." %in% dev_on_arg_names
    if (length(dots_pos) == 0) stop("`dev_on()` must contain arguments `...` if using `req`")
    req_pos <- which("..." %in% dev_on_arg_names
    if (!isTRUE(dots_pos < req_pos)) {
      stop("`dev_on()` must have arguments `...` before `req=`)
    }
  }
```

Reply:
Thanks for all the immediate feedback...

Given how much there is, it might be an idea to make and edit these changes in an editor and to add some unit tests as we go?

If it helps, we can hold off on this (and the other PRs) until RStudio have capacity to invest the time? (That might also help me negotiate some wriggle room with my work pressures too!)